scripts/create_sfm_from_obj.py

```diff
@@ -0,0 +1,145 @@
+import os
+import sys
+import argparse
+from pathlib import Path
+import numpy as np
+import pycolmap
+import cv2
+from tqdm import tqdm
+from pixtrack.utils.pytorch3d_render_utils import (create_look_at_poses_for_mesh, 
+                                                   render_image,
+                                                   create_colmap_camera,
+                                                   create_colmap_image_from_pytorch3d_RT)
+from hloc import (extract_features, 
+                  match_features, 
+                  reconstruction, 
+                  pairs_from_exhaustive)
+import hloc.triangulation as triangulation
+from hloc.triangulation import (run_triangulation, 
+                                create_db_from_model,
+                                import_features, 
+                                import_matches, 
+                                geometric_verification)
+from hloc.reconstruction import (create_empty_db, 
+                                 import_images, 
+                                 get_image_ids)
+from hloc.utils.read_write_model import (write_images_binary, 
+                                         write_points3D_binary,
+                                         write_cameras_binary)
+
+
+def render_dataset_and_get_colmap_images_and_cameras(mesh_path, dataset_path, 
+                                                     fx, fy, cx, cy, W, H):
+    mesh_path = args.mesh_path
+    Rs, Ts, mesh = create_look_at_poses_for_mesh(mesh_path, subdivisions=2)
+    dataset_path.mkdir(parents=True, exist_ok=True)
+    mapping_path = dataset_path / 'mapping'
+    mapping_path.mkdir(parents=True, exist_ok=True)
+    colmap_images = {}
+    colmap_camera_id = 1
+    colmap_camera = create_colmap_camera(W, H, fx, cx, cy, k1=0.)
+    colmap_cameras = {colmap_camera_id: colmap_camera}
+    for i in tqdm(range(Rs.shape[0])):
+        R = Rs[i].unsqueeze(0)
+        T = Ts[i].unsqueeze(0)
+        image = render_image(mesh, fx, fy, cx, cy, W, H, R, T)
+        image = (image[..., :3] * 255.).astype(np.uint8)
+        image = cv2.cvtColor(image, cv2.COLOR_RGB2BGR)
+        img_name = ('%d.png' % (i + 1)).zfill(8)
+        cv2.imwrite(str(mapping_path / img_name), image)
+
+        colmap_image_name = 'mapping/' + img_name
+        colmap_image_id = i + 1
+        colmap_image = create_colmap_image_from_pytorch3d_RT(R[0], T[0], 
```

Comment:
Indexing for `0` is hard to interpret. Can you make it functional instead?

Reply:
Refactored the code